pkg/discovery/processor/controller_processor.go

```diff
@@ -0,0 +1,116 @@
+package processor
+
+import (
+	"strings"
+
+	"github.com/golang/glog"
+
+	"github.com/davecgh/go-spew/spew"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/turbonomic/kubeturbo/pkg/cluster"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	"github.com/turbonomic/kubeturbo/pkg/util"
+)
+
+// Query and cache predefined controllers
+var (
+	supportedControllers = []schema.GroupVersionResource{
+		{
+			Group:    util.K8sAPIReplicationControllerGV.Group,
+			Version:  util.K8sAPIReplicationControllerGV.Version,
+			Resource: util.ReplicationControllerResName,
+		},
+		{
+			Group:    util.K8sAPIReplicasetGV.Group,
+			Version:  util.K8sAPIReplicasetGV.Version,
+			Resource: util.ReplicaSetResName,
+		},
+		{
+			Group:    util.K8sAPIDeploymentGV.Group,
+			Version:  util.K8sAPIDeploymentGV.Version,
+			Resource: util.DeploymentResName,
+		},
+		{
+			Group:    util.OpenShiftAPIDeploymentConfigGV.Group,
+			Version:  util.OpenShiftAPIDeploymentConfigGV.Version,
+			Resource: util.DeploymentConfigResName,
+		},
+		{
+			Group:    util.K8sAPIStatefulsetGV.Group,
+			Version:  util.K8sAPIStatefulsetGV.Version,
+			Resource: util.StatefulSetResName,
+		},
+		{
+			Group:    util.K8sAPIDaemonsetGV.Group,
+			Version:  util.K8sAPIDaemonsetGV.Version,
+			Resource: util.DaemonSetResName,
+		},
+	}
+)
```

Comment:
Are you intentionally ignoring "Job" and "CronJob" here? They don't have replicas but looks they could have labels (https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job). I'm wondering if we want to include "Job" and "CronJob" as well to facilitate the task Enlin will be working on.

Reply:
Good catch @marvelyue . @ading1977 please add the Job and CronJob.

Reply:
Added.